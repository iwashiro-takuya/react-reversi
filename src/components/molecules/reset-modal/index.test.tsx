import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Player from 'src/const/player';
import ResetModal from '.';
import useResetModal from './hooks';

jest.mock('./hooks');
const useResetModalMock = useResetModal as jest.MockedFunction<
  typeof useResetModal
>;

afterEach(cleanup);

describe('ResetModal', () => {
  it(`snapshot(${Player.PLAYER_1.name})`, () => {
    useResetModalMock.mockReturnValue({
      onReset: () => {
        return true;
      },
      gameResultText: `${Player.PLAYER_1.name} win!`,
    });
    const { asFragment } = render(<ResetModal />);
    expect(asFragment()).toMatchSnapshot();
  });

  it(`snapshot(${Player.PLAYER_2.name})`, () => {
    useResetModalMock.mockReturnValue({
      onReset: () => {
        return true;
      },
      gameResultText: `${Player.PLAYER_2.name} win!`,
    });
    const { asFragment } = render(<ResetModal />);
    expect(asFragment()).toMatchSnapshot();
  });

  it(`snapshot(Draw)`, () => {
    useResetModalMock.mockReturnValue({
      onReset: () => {
        return true;
      },
      gameResultText: `Draw!`,
    });
    const { asFragment } = render(<ResetModal />);
    expect(asFragment()).toMatchSnapshot();
  });
});
