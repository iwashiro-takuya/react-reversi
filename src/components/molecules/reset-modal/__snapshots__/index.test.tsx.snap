// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetModal snapshot(Draw) 1`] = `
<DocumentFragment>
  <div
    class="sc-gtsrHT iBzBKB"
  >
    <div
      class="sc-dlnjwi laUVez"
    >
      <p
        class="sc-hKFxyN bnINoU"
      >
        Game Result
      </p>
      <p
        class="sc-eCApnc krybIm"
      >
        Draw!
      </p>
      <button
        class="sc-bdnxRM jTgbIw"
        data-cy="reset"
        type="submit"
      >
        Game Reset
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ResetModal snapshot(Player 1) 1`] = `
<DocumentFragment>
  <div
    class="sc-gtsrHT iBzBKB"
  >
    <div
      class="sc-dlnjwi laUVez"
    >
      <p
        class="sc-hKFxyN bnINoU"
      >
        Game Result
      </p>
      <p
        class="sc-eCApnc krybIm"
      >
        Player 1 win!
      </p>
      <button
        class="sc-bdnxRM jTgbIw"
        data-cy="reset"
        type="submit"
      >
        Game Reset
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ResetModal snapshot(Player 2) 1`] = `
<DocumentFragment>
  <div
    class="sc-gtsrHT iBzBKB"
  >
    <div
      class="sc-dlnjwi laUVez"
    >
      <p
        class="sc-hKFxyN bnINoU"
      >
        Game Result
      </p>
      <p
        class="sc-eCApnc krybIm"
      >
        Player 2 win!
      </p>
      <button
        class="sc-bdnxRM jTgbIw"
        data-cy="reset"
        type="submit"
      >
        Game Reset
      </button>
    </div>
  </div>
</DocumentFragment>
`;
